server/src/main/java/io/crate/session/Session.java

```diff
@@ -460,6 +474,10 @@ public DescribeResult describe(char type, String portalOrStatement) {
         }
     }
 
+    protected long phaseTime(String phase, long phaseStart) {
```

Comment:
Hm not sure - the whole point of this was to imitate time of some phases. 

I tried to move it to the token and realized that I have to 
- mock token (use fake times)
- mock session (use mocked token)
- token creation is encapsulated in `Session.parse`, would have to expose it for for testing, so that we can control that we create mock token.

So 3 changes instead of 1, not sure it worth it

Reply:
I will update tests, implementation is quite different after applying suggestion below